### PR TITLE
Feature: Added run as admin toggle to shortcut properties 

### LINKED ISSUE #4394: Feature: Add an option to the properties window to run shortcuts as admin
- [ ] Add a toggle switch to the shortcut properties page to run shortcuts as administrator.

### PR DESCRIPTION
**Pull request by hecksmosis**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #4394 

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [x] Tested the changes for accessibility

### CODE REVIEW

**Comment by hecksmosis** [comment]
I am making this a draft because the refresh function is quite suboptimal (it refreshes everything on the page), I'm going to look for a solution but help would be appreciated.

File: src/Files.App/Strings/en-GB/Resources.resw [hunk 1]
```diff
@@ -2935,4 +2940,10 @@
   <data name="SortBy" xml:space="preserve">
     <value>Sort by</value>
   </data>
+  <data name="PropertiesDialogIsRunAsAdmin.Text" xml:space="preserve">
+    <value>Run as administrator</value>
+  </data>
+  <data name="RunAsAdmin" xml:space="preserve">
+    <value>Run as administrator</value>
+  </data>
```
Comment (yaira2) [inline src/Files.App/Strings/en-GB/Resources.resw:2948]: Can you revert the changes to this file? Translations are setup to sync from Crowdin.

Comment (hecksmosis) [reply src/Files.App/Strings/en-GB/Resources.resw:2948]: Oh ok

**Comment by QuaintMako** [comment]
Could you fix the formatting that was modified by accident but has nothing to do with the PR ? Like in `src/Files.App/Views/Pages/PropertiesGeneral.xaml` so it'd be easier to review what really needs to be? Thanks in advance!

**Comment by hecksmosis** [comment]
Yes, idk why but my formatting turns strange when I push changes to github

**Comment by gave92** [comment]
I think this should go under the "Shortcut" tab in properties instead of general properties.

File: src/Files.App/Views/Pages/PropertiesGeneral.xaml [hunk 2]
```diff
@@ -534,29 +536,52 @@
...
+						Grid.Column="0"
+						Grid.ColumnSpan="3"
+						Margin="-12,0"
+						Background="{ThemeResource CardStrokeColorDefaultBrush}" />
+
+                    <TextBlock
+						Grid.Row="4"
+						Grid.Column="0"
+						VerticalAlignment="Center"
+						Text="{helpers:ResourceString Name=PropertiesDialogIsRunAsAdmin/Text}"
```
Comment (gave92) [inline src/Files.App/Views/Pages/PropertiesGeneral.xaml:572]: We can use the same string resource ("RunAsAdmin") for both.

**Comment by hecksmosis** [comment]
It is already disabled in folder shortcuts

File: src/Files.App/Views/Pages/PropertiesGeneral.xaml.cs [hunk 3]
```diff
@@ -104,7 +105,30 @@ async Task<bool> SaveBaseAsync()
...
+				if (item.IsShortcut)
+				{
+					var shortcutItem = (ShortcutItem)item;
+
+					App.Logger.Warn("Is a shortcut file");
+
+					 var isApplication = !string.IsNullOrWhiteSpace(shortcutItem.TargetPath) &&
+						(shortcutItem.TargetPath.EndsWith(".exe", StringComparison.OrdinalIgnoreCase)
+						|| shortcutItem.TargetPath.EndsWith(".msi", StringComparison.OrdinalIgnoreCase)
+						|| shortcutItem.TargetPath.EndsWith(".bat", StringComparison.OrdinalIgnoreCase));
```
Comment (yaira2) [inline src/Files.App/Views/Pages/PropertiesGeneral.xaml.cs:119]: This would be a good candidate for #10831. @QuaintMako

**Comment by gave92** [comment]
Enabled for shortcuts to folder:
image

Enabled for shortcuts to txt:
image

**Comment by hecksmosis** [comment]
I'll fix it tomorrow

File: src/Files.App/Views/Pages/PropertiesGeneral.xaml.cs (hunk 3 above)
Comment (hecksmosis) [reply src/Files.App/Views/Pages/PropertiesGeneral.xaml.cs:119]: Yes, it would be nice to have an extension method for IsExecutable

Comment (hecksmosis) [reply src/Files.App/Views/Pages/PropertiesGeneral.xaml.cs:119]: Should I implement it in this pr?

Comment (yaira2) [reply src/Files.App/Views/Pages/PropertiesGeneral.xaml.cs:119]: @gave92 opened a PR for this so let's wait for that to be merged and then you can take advantage of his changes.

Comment (hecksmosis) [reply src/Files.App/Views/Pages/PropertiesGeneral.xaml.cs:119]: Ok

File: src/Files.App/ViewModels/Properties/FileProperties.cs [hunk 4]
```diff
@@ -114,7 +115,17 @@ public override async void GetSpecialProperties()
 				ViewModel.ItemCreatedTimestamp = Item.ItemDateCreated;
 				ViewModel.ItemAccessedTimestamp = Item.ItemDateAccessed;
 				ViewModel.LoadLinkIcon = Item.LoadWebShortcutGlyph;
-                ViewModel.IsRunAsAdmin = ((ShortcutItem)Item).RunAsAdmin;
+
+				var shortcutItem = (ShortcutItem)Item;
+
+                var isApplication = !string.IsNullOrWhiteSpace(shortcutItem.TargetPath) &&
```
Comment (gave92) [inline src/Files.App/ViewModels/Properties/FileProperties.cs:121]: I think that putting this in GetBaseProperties() is enough. No need to do this twice.

Comment (hecksmosis) [reply src/Files.App/ViewModels/Properties/FileProperties.cs:121]: Oh true

**Comment by hecksmosis** [comment]
Should I try to change the refresh function? (As it is not in the main page any more, I find it unlikely that many people will change this repeatedly, and therefore it will not have a significant effect on performance

**Comment by gave92** [comment]
Hi I've removed an extra call to CreateOrUpdateLinkAsync(), and formatted using tabs instead of spaces.
Also I've removed the call to RefreshItems(), setting `item.RunAsAdmin = isRunAsAdmin;` seems enough for me.

**Comment by hecksmosis** [comment]
Ok, sounds good. Should I mark this as ready for review?

File: src/Files.App/Strings/en-US/Resources.resw [hunk 5]
```diff
@@ -2935,4 +2940,7 @@
   <data name="SortBy" xml:space="preserve">
     <value>Sort by</value>
   </data>
+  <data name="RunAsAdmin" xml:space="preserve">
+    <value>Run as administrator</value>
```
Comment (yaira2) [inline src/Files.App/Strings/en-US/Resources.resw:2944]: Isn't there already a string for this being used in the compatibility page?

Comment (hecksmosis) [reply src/Files.App/Strings/en-US/Resources.resw:2944]: Yes, I think we should change that one because RunAsAdmin is shorter.

**Comment by yaira2** [comment]
Turning the toggle off sometimes requires multiple clicks before it works.

**Comment by hecksmosis** [comment]
It's weird, I cannot reproduce this consistently.

**Comment by hecksmosis** [comment]
@yair100 This looks like an issue with the xaml component, as it also happens in parts of the settings page with the same component.

**Comment by yaira2** [comment]
Which setting?

**Comment by hecksmosis** [comment]
The ones that use a ToggleSwitch occasionally have this issue

**Comment by hecksmosis** [comment]
It looks like a known issue:
https://github.com/microsoft/microsoft-ui-xaml/issues/3796

File: src/Files.App/ViewModels/SelectedItemsPropertiesViewModel.cs [hunk 6]
```diff
@@ -602,5 +602,23 @@ public bool IsHidden
...
+			get => runAsAdmin;
+			set
+			{
+				RunAsAdminEnabled = true;
+				SetProperty(ref runAsAdmin, value);
+			}
+		}
+
+		private bool runAsAdminEnabled;
+		public bool RunAsAdminEnabled
```
Comment (yaira2) [inline src/Files.App/ViewModels/SelectedItemsPropertiesViewModel.cs:618]: What's this value for?

Comment (gave92) [reply src/Files.App/ViewModels/SelectedItemsPropertiesViewModel.cs:618]: Enabling the checkbox

Comment (yaira2) [reply src/Files.App/ViewModels/SelectedItemsPropertiesViewModel.cs:618]: Should probably be used to hide the toggle switch?

Comment (gave92) [reply src/Files.App/ViewModels/SelectedItemsPropertiesViewModel.cs:618]: I think it was done to match the "read only" and "hidden" that get disabled and not hidden. But since the toggle has been moved to Shortut page perhaps it's better to hide it.

Comment (hecksmosis) [reply src/Files.App/ViewModels/SelectedItemsPropertiesViewModel.cs:618]: Done

File: src/Files.App/Strings/en-US/Resources.resw [hunk 7]
```diff
@@ -1,15 +1,17 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <root>
-  <!--
-    Microsoft ResX Schema
-
+  <!--
```
Comment (yaira2) [inline src/Files.App/Strings/en-US/Resources.resw:3]: Can you revert changes to spacing?

Comment (hecksmosis) [reply src/Files.App/Strings/en-US/Resources.resw]: Done

File: src/Files.App/ViewModels/SelectedItemsPropertiesViewModel.cs [hunk 8]
```diff
@@ -602,5 +602,23 @@ public bool IsHidden
 			get => isHidden;
 			set => SetProperty(ref isHidden, value);
 		}
+
+		private bool runAsAdmin;
+		public bool RunAsAdmin
```
Comment (yaira2) [inline src/Files.App/ViewModels/SelectedItemsPropertiesViewModel.cs:607]: Can you add docs to the properties in this ViewModel?

Comment (hecksmosis) [reply src/Files.App/ViewModels/SelectedItemsPropertiesViewModel.cs:607]: What do you mean?

Comment (yaira2) [reply src/Files.App/ViewModels/SelectedItemsPropertiesViewModel.cs:607]: Comments

Comment (hecksmosis) [reply src/Files.App/ViewModels/SelectedItemsPropertiesViewModel.cs:607]: Ok

Comment (hecksmosis) [reply src/Files.App/ViewModels/SelectedItemsPropertiesViewModel.cs:607]: Should I finally do this? (It's not a problem, just confirming after what d2dyno said)

Comment (yaira2) [reply src/Files.App/ViewModels/SelectedItemsPropertiesViewModel.cs:607]: This is good as is 👍

**Review by yair100 — DISMISSED** [review]
LGTM

**Review by QuaintMako — DISMISSED** [review]

**Comment by hecksmosis** [comment]
Is this ok then?

**Review by yair100 — APPROVED** [review]
LGTM
